### DISCUSSION TITLE
Configuration for QField's new attribute form features (version 3.3)

### DISCUSSION THREAD

**Discussion by lorenzomorning** [opengisch/QField#5433] [Q&A]
_Note: I am using QGIS 3.28.15 (on a Windows computer) and QField 3.3.6 (on a Windows based Tablet)._

I am wondering how/where to enable the (pretty cool) following new features of the new release 3.3: 

> - The feature form’s value relation widget gained grouping capabilities to enhance ordering of items
> - The feature form’s value map widget gained a toggle buttons interface for faster data entry and an item filtering for large combobox lists
![grafik](https://github.com/opengisch/QField/assets/45905290/e2d07534-e677-47a7-9496-c95574bb3bcd)

From my understanding of QField such features like the layout/settings of the attribute form are configured in the QGIS project itself. However in my current QGIS version 3.28 I cannot figure out where to enable in example the toggle buttons for the value map widget. 
![grafik](https://github.com/opengisch/QField/assets/45905290/54a61f00-a510-48d5-bd6a-f01351872052)

Therefore, my questions: Are these new settings configured in QGIS? What version of QGIS would I need for the configuration?

**Comment by nirvn** [comment]
@lorenzomorning , it's a layer-wide setting, you'll find it in the QField panel (provided you installed the QFieldSync plugin):

![image](https://github.com/opengisch/QField/assets/1728657/e6d3fa31-cec6-480f-b5e5-74055da2bf2c)

**Reply by lorenzomorning** [reply]
@nirvn thanks for the fast response and hint! The layer-wide setting totally makes sense 👍 
However, the newest version of QFieldSync that is available at my Plugin Manager in QGIS is v.4.9.1 
![grafik](https://github.com/opengisch/QField/assets/45905290/a0bfdfa4-d7df-486c-a457-ac8e8fe53f13)
But your [release changelog of QFieldSync](https://github.com/opengisch/qfieldsync/releases) indicates that the configuration of the new toggle buttons is only available from v.4.10.0 on.
So must I install the newest version of the plugin via the source code from a .zip file or is the installation somehow directly accessble via the QGIS Plugin Manager? 

_Note: When I install the v.4.10.0 from the .zip file provided on github, unfortunatley my layer properties look like the following:_ 
![grafik](https://github.com/opengisch/QField/assets/45905290/08453857-7159-4ff9-9d33-9cb3817a5d2b)

_with the following error message:_ 
```
AttributeError: 'LayerSource' object has no attribute 'value_map_button_interface_threshold' 
Traceback (most recent call last):
  File "C:\Users/lorenz.beck/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qfieldsync\gui\map_layer_config_widget.py", line 48, in createWidget
    return MapLayerConfigWidget(layer, canvas, parent)
  File "C:\Users/lorenz.beck/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qfieldsync\gui\map_layer_config_widget.py", line 131, in __init__
    self.layer_source.value_map_button_interface_threshold
AttributeError: 'LayerSource' object has no attribute 'value_map_button_interface_threshold'


Python-Version: 3.9.18 (heads/master:c290dc6, Jan 14 2024, 01:36:02) [MSC v.1929 64 bit (AMD64)] 
QGIS-Version: 3.28.15-Firenze Firenze, 252ad49ddc 
```

**Reply by lorenzomorning** [reply]
Hi @nirvn,
Is it planned to make the the newer version (4.10.0) of QFieldSync available via the Plugin Manager for more "older" QGIS versions? :) 
I am currently working with QGIS 3.28.15 and unfortunately the newest available QFieldSync version (4.9.1) does not allow me to set the value map toggle buttons.

**Reply by SeqLaz** [reply]
Hi @lorenzomorning Could you try uninstalling and installing again?

**Reply by lorenzomorning** [reply]
Hi @SeqLaz, 

I uninstalled and reinstalled the plugin, but nothing changes in the user interface. As you can see there is no additional option to set the toggle buttons for a value map:

![grafik](https://github.com/user-attachments/assets/b349c240-c345-46a9-87e5-9c9d001d00ff)

**Reply by SeqLaz** [reply]
Hi @lorenzomorning, when uninstalling and installing it again, did you use the zipped version [4.10.0](https://github.com/opengisch/qfieldsync/releases/tag/v4.10.0) or from the plugins repository?

**Reply by lorenzomorning** [reply]
Hi @SeqLaz,
I now tried both: 

1. Installing it from the plugins repository (4.9.1) obviously does not add any new layer control sections (see image above)
2. Installing the zipped version [4.10.0](https://github.com/opengisch/qfieldsync/releases/tag/v4.10.0) results in a kind of overlay for the QField layer settings (see image below, or [post from 9th July](https://github.com/opengisch/QField/discussions/5433#discussioncomment-9997472)). It now adds the option for controlling the toggle buttons of a value map, but unfortunately it is not nicely usable due the overlay...

![grafik](https://github.com/user-attachments/assets/1f2e6e73-4b2d-42b0-b03a-a355041b378c)

**Reply by lorenzomorning** [reply]
Hi @SeqLaz @nirvn,
you are probably super buisy and I do not need a direct solution for my problem described above. But it is planned to soon integrate the current QFieldSync updates (e.g. toggle button control) into the version available within the QGIS Plugin Manager? :)

**Reply by SeqLaz** [reply]
Hi @lorenzomorning,

Could be possible from your side to upgrade the QGIS version to the latest version?

**Reply by m-kuhn** [reply]
Release of qfieldsync 4.10 has been delayed due to technical problems with the upload to the plugin registry, but has finally made it today. [v4.10.1](https://github.com/opengisch/qfieldsync/releases/tag/v4.10.1) is now available from the QGIS plugin repository and comes with this feature.

**Reply by lorenzomorning** [reply]
@m-kuhn , @nirvn , @SeqLaz thank you all for the update and the great and steady improvements of QField. It is so valuable for us working with this software, and at the same time it's a lot of fun :)
